MFDecodeH264/MFDecodeH264.cpp

```diff
@@ -1,147 +1,581 @@
+            else
+            {
+                _RPTWN(_CRT_WARN, L"Draining, TotalRead %u\n", TotalRead);
+                // end of file, ask decoder to process all data from previous calls
+                HRCHECK(decoder->ProcessMessage(MFT_MESSAGE_COMMAND_DRAIN, 0));
+            }
+            TotalRead += read;
+
+            for(bool Continue = true; Continue; )
+            {
+
+            MFT_OUTPUT_STREAM_INFO output_stream_info;
+            HRESULT hrtemp = decoder->GetOutputStreamInfo(0, &output_stream_info);
+            WINRT_ASSERT(output_stream_info.cbSize);
+
+            winrt::com_ptr<IMFSample> outputSample;
+            HRCHECK(MFCreateSample(outputSample.put()));
+            MFT_OUTPUT_DATA_BUFFER outputBuffer{};
+            outputBuffer.pSample = outputSample.get();
+
+            //if (sampleSize)
+            {
+                // now we know the size so we can (and must) allocate the MF output buffer
+                winrt::com_ptr<IMFMediaBuffer> outputBuffer;
+                HRCHECK(MFCreateMemoryBuffer(output_stream_info.cbSize, outputBuffer.put()));
+                HRCHECK(outputSample->AddBuffer(outputBuffer.get()));
+            } // else just continue to process
+
+            DWORD status = 0;
+            auto hr = decoder->ProcessOutput(0, 1, &outputBuffer, &status);
+            _RPTWN(_CRT_WARN, L"hr 0x%08X\n", hr);
+
+            if (hr == MF_E_TRANSFORM_NEED_MORE_INPUT) // just go on
+            {
+                if (!read) // file is all read
+                {
+                    Done = true;
+                    break;
+                }
+
+                break; //continue;
+            }
+            
+            // https://learn.microsoft.com/en-us/windows/win32/medfound/handling-stream-changes
+            if (hr == MF_E_TRANSFORM_STREAM_CHANGE)
+            {
+                // get (and set) NV12 output type (could be I420, IYUV, YUY2, YV12)
+                HRCHECK(SetOutputType(decoder, MFVideoFormat_NV12));
+                HRESULT hrtemp = decoder->GetOutputStreamInfo(0, &output_stream_info);
+                bool test = StreamAllocatesMemory(output_stream_info.dwFlags);
+                // now get the sample size
+                winrt::com_ptr<IMFMediaType> type;
+                HRCHECK(decoder->GetOutputCurrentType(0, type.put()));
+                HRCHECK(type->GetUINT32(MF_MT_SAMPLE_SIZE, &sampleSize));
+                break; //continue;
+            }
+            HRCHECK(hr);
+
+            LONGLONG time, duration;
+            HRCHECK(outputSample->GetSampleTime(&time));
+            HRCHECK(outputSample->GetSampleDuration(&duration));
+            wprintf(L"Sample time: %I64u ms duration: %I64u ms\n", time / 10000, duration / 10000);
+            c++; // need 250, but now 230,234,240
+
+            }
+
+            notAccept;
+        }
+
+        WINRT_ASSERT(TotalRead == 1'280'742u);
+        wprintf(L"Total cound %d\n", c);
+        WINRT_ASSERT(c == 250u);

```

Comment:
However you might see that sucking output from the MFT is still not always working, but this time c will be 249 (or maybe 248) from time to time. Perhaps more rarely than you originally saw.